File: Problems/No.1827_Minimum_Operations_to_Make_the_Array_Increasing/minimum_operations_to_make_the_array_increasing.py
class Solution(object):
    def minOperations(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # parameter
        operations = 0

        # start computing
        for i in range(1, len(nums)):
            if (nums[i-1] >= nums[i]):
                operations += (nums[i-1] - nums[i] + 1)
                nums[i] = nums[i-1] + 1
        
        return operations

File: Problems/No.1827_Minimum_Operations_to_Make_the_Array_Increasing/minimum_operations_to_make_the_array_increasing.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        
        // parameter
        int operations = 0;

        // start computing
        for (int i=1; i<nums.size(); i++) 
            if (nums[i-1] >= nums[i]) {

                operations += (nums[i-1] -nums[i] + 1);
                nums[i] = nums[i-1] + 1;

            }
        
        return operations;

    }
};
